src/services/platform.service.js

```diff
@@ -72,33 +72,38 @@ export const getBaseProjectEnvironment = (
   };
 };
 
-window.childProcess = childProcess;
-
 // HACK: With electron-builder, we're having some issues on mac finding Node.
 // This is because for some reason, the PATH is not updated properly :(
 // 'fix-path' is supposed to do this for us, but it doesn't work, for unknown
 // reasons.
 export const initializePath = () => {
-  childProcess.exec('which node', { env: window.process.env }, (_, version) => {
-    if (!version && isMac) {
-      // For users with a standard Node installation, node will be in
-      // /usr/local/bin
-      // For users using NVM, the path to Node will be added to `.bashrc`.
-      // Add both to the PATH.
-      try {
-        childProcess.exec(
-          'source ~/.bashrc && echo $PATH',
-          (err, updatedPath) => {
-            if (updatedPath) {
-              window.process.env.PATH = `/usr/local/bin:${updatedPath}`;
-            }
+  return new Promise((resolve, reject) => {
+    childProcess.exec(
+      'which node',
+      { env: window.process.env },
+      (_, version) => {
+        if (!version && isMac) {
```

Comment:
if there is a version or it's not a mac, it will never resolve

Reply:
Good catch. I think we could move the `isMac` check before `which node` `exec` and `resolve` in the else case as this is only required on Mac. Also `which` command could be a problem on Windows as it's called `where` there.

I think also for `!version` we need the else-case to `resolve(version)`. By the way, `version` is the `path` to node and not the version as `which node` returns a path - so we could rename it to something like `nodePath`.

I think we shouldn't reject inside `initializePath` as this will skip the node check that follows afterwards. So always resolving should be OK.